Ah and `tests/models/git/test_modeling_git.py::GitModelTest::test_resize_tokens_embeddings - AssertionError: Padding_idx must be within num_embeddings` as well

--- a/src/transformers/modeling_utils.py
+++ b/src/transformers/modeling_utils.py
@@ -2447,7 +2447,7 @@ class PreTrainedModel(nn.Module, ModuleUtilsMixin, GenerationMixin, PushToHubMix
         if is_covariance_psd:
             # If covariances is positive definite, a distribution can be created. and we can sample new weights from it.
             distribution = torch.distributions.multivariate_normal.MultivariateNormal(
-                mean_embeddings, covariance_matrix=1e-5 * covariance
+                mean_embeddings, covariance_matrix=1e-7 * covariance
             )
             new_embeddings.weight.data[-1 * added_num_tokens :, :] = distribution.sample(
                 sample_shape=(added_num_tokens,)
@@ -2485,7 +2485,7 @@ class PreTrainedModel(nn.Module, ModuleUtilsMixin, GenerationMixin, PushToHubMix
     def _init_added_lm_head_bias_with_mean(self, old_lm_head, new_lm_head, added_num_tokens):
         bias_mean = torch.mean(old_lm_head.bias.data, axis=0, dtype=torch.float32)
         bias_std = torch.std(old_lm_head.bias.data, axis=0).to(torch.float32)
-        new_lm_head.bias.data[-1 * added_num_tokens :].normal_(mean=bias_mean, std=bias_std * 1e-5)
+        new_lm_head.bias.data[-1 * added_num_tokens :].normal_(mean=bias_mean, std=1e-7 * bias_std)
 
     def _copy_lm_head_original_to_resized(
         self, new_lm_head, old_lm_head, num_tokens_to_copy, transposed, has_new_lm_head_bias

--- a/tests/test_modeling_common.py
+++ b/tests/test_modeling_common.py
@@ -2030,7 +2030,7 @@ class ModelTesterMixin:
                 old_embeddings_mean = torch.mean(output_embeds.weight.data[:-10, :], axis=0)
                 new_embeddings_mean = torch.mean(output_embeds.weight.data[-10:, :], axis=0)
             torch.testing.assert_close(old_embeddings_mean, new_embeddings_mean, atol=1e-3, rtol=1e-1)
-            # check if the bias is always initialized with zero.
+            # check if the old bias mean close to added bias mean.
             if output_embeds.bias is not None:
                 if is_deepspeed_zero3_enabled():
                     with deepspeed.zero.GatheredParameters(output_embeds.bias, modifier_rank=None):

--- a/tests/test_modeling_common.py
+++ b/tests/test_modeling_common.py
@@ -1886,6 +1886,9 @@ class ModelTesterMixin:
             self.assertTrue(models_equal)
 
             del model
+            del config
+            # Copy again. config changed with embedding resizing (`vocab_size` changed)
+            config = copy.deepcopy(original_config)
             if is_deepspeed_zero3_enabled():
                 with deepspeed.zero.Init():
                     model = model_class(config)
@@ -1922,6 +1925,7 @@ class ModelTesterMixin:
             # Test when `vocab_size` is smaller than `hidden_size`.
             del model
             config.vocab_size = 4
+            config.pad_token_id = 3
             if is_deepspeed_zero3_enabled():
                 with deepspeed.zero.Init():
                     model = model_class(config)

